### PR TITLE
Consolidate batchSystem tests (resolves #354)

### LINKED ISSUE #354: Improve batch system tests
Make them faster and reduce coverage overlap between individual test methods.

### PR DESCRIPTION
**Pull request by adderan**
- Replace the batchSystem tests that involve issuing jobs with a single test that uses issueBatchJob, getIssuedBatchJobIDs, getRunningBatchJobIDs, getUpdatedBatchJobIDs, and killBatchJobs together.
- Don't let jobs go to sleep for arbitrary lengths of time - make the running time of the test be limited only by how fast the batch system can run the API calls. 
- Avoid race conditions that occur when getRunningJobIDs() or getIssuedJobIDs() are called after wait_for_jobs(). Using two batchSystem API calls was leaving a window open for the job to change status before the second call.

### CODE REVIEW

**Comment by adderan** [comment]
This seems to be working now.

File: src/toil/test/batchSystems/batchSystemTest.py [hunk 1]
```diff
@@ -82,13 +86,40 @@ def tearDown(self):
...
+        def testRunJobs(self):
+            testPath = os.path.join(self.tempDir, "test.txt")
+
+            job1 = self.batchSystem.issueBatchJob("sleep 1000",
+                                                  memory=memoryForJobs, cores=0.5, disk=diskForJobs)
+            job2 = self.batchSystem.issueBatchJob("sleep 1000",
+                                                  memory=memoryForJobs, cores=0.5, disk=diskForJobs)
+
+            issuedIDs = self.waitForJobsToIssue(2)
+            self.assertEqual(set(issuedIDs), set([job1, job2]))
```
Comment (hannes-ucsc) [inline src/toil/test/batchSystems/batchSystemTest.py]: {job1, job2}

File: src/toil/test/batchSystems/batchSystemTest.py [hunk 2]
```diff
@@ -82,13 +86,40 @@ def tearDown(self):
...
+
+            runningJobIDs = self.waitForJobsToStart(2)
+            self.assertEqual(set(runningJobIDs), set([job1, job2]))
```
Comment (hannes-ucsc) [inline src/toil/test/batchSystems/batchSystemTest.py]: Same here.

**Comment by hannes-ucsc** [comment]
I just did some minor clean-up, rebased and force-pushed.

**Comment by hannes-ucsc** [comment]
Jenkins slave was messed up. Trying again.

**Comment by hannes-ucsc** [comment]
Jenkins, test this please.

**Pull request merged**